process/block/preprocess/smartContractResults_test.go

```diff
@@ -649,12 +640,41 @@ func TestScrsPreprocessor_ProcessMiniBlock(t *testing.T) {
 
 }
 
-func TestScrs_Preprocess_RestoreTxBlockIntoPools(t *testing.T) {
+func TestScrsPreprocessor_ProcessMiniBlockWrongTypeMiniblockShouldErr(t *testing.T) {
+	t.Parallel()
+	tdp := initDataPool()
+	requestTransaction := func(shardID uint32, txHashes [][]byte) {}
+	scr, _ := NewSmartContractResultPreprocessor(
+		tdp.Transactions(),
+		&mock.ChainStorerMock{},
+		&mock.HasherMock{},
+		&mock.MarshalizerMock{},
+		&mock.TxProcessorMock{},
+		mock.NewMultiShardsCoordinatorMock(3),
+		&mock.AccountsStub{},
+		requestTransaction,
+	)
+	body := make(block.Body, 0)
+	txHash := []byte("tx1_hash")
+	txHashes := make([][]byte, 0)
+	txHashes = append(txHashes, txHash)
+	miniblock := block.MiniBlock{
+		ReceiverShardID: 0,
+		SenderShardID:   0,
+	}
+	body = append(body, &miniblock)
+	err := scr.ProcessMiniBlock(&miniblock, func() bool { return true }, 1)
```

Comment:
you can create a function for haveTime and call it when you need it instead of defining a new function for every test which requires it

Reply:
Done.